packages/shared/src/hooks/useTagAndSource.ts

```diff
@@ -22,7 +23,7 @@ export interface SourceActionArguments {
 }
 
 interface UseTagAndSourceProps {
-  origin?: string;
+  origin?: AuthTriggers;
```

Comment:
This should stay Origin ideally.
If it's used as a Trigger we can say `Origin as Trigger` in the implementation needed